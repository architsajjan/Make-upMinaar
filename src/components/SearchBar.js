import React from 'react';

export default function SearchBar(props) {
    return (
        <div>
            <form className="form searchForm form-bg">
                <div className="form-group">
                    <input type="text" className="SearchBox" spellCheck="false"  placeholder="Search for a product/ brand/ category" name="searchedText" onChange={props.handleChange}  required />
                </div>
                <div className="form-group">
                    <input type="button" className="btn primay-btn SearchButton"  value="Search"  onClick={props.searchItem}/>
                </div>
            </form>
        </div>
    )
}
