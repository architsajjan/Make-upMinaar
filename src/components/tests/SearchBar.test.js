import React from "react";
import { render, unmountComponentAtNode } from "react-dom";

import SearchBar from '../SearchBar';

it("runs without crash", () => {
  const div = document.createElement("div");
  render(<SearchBar 
          searchItem={()=>{}} 
          handleChange={()=>{}}
          autocompleteCallback = {()=>{}} 
          names={[]}
          brands={[]}
          categories={[]}
          types={[]}
          tags={[]}
        />
        , div);
  unmountComponentAtNode(div);
});