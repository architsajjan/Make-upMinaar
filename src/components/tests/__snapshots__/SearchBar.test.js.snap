// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <form
    className="form searchForm form-bg"
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <input
        autoComplete="off"
        className="SearchBox"
        name="searchedText"
        onChange={[Function]}
        placeholder="Search for a product/ brand/ category"
        spellCheck="false"
        type="text"
      />
    </div>
    <div
      className="form-group"
    >
      <input
        className="btn primay-btn SearchButton"
        type="submit"
        value="All Products"
      />
    </div>
  </form>
</div>
`;
